pydomcfg/domzgr/zco.py

```diff
@@ -0,0 +1,294 @@
+#!/usr/bin/env python
+
+"""
+Class to generate NEMO v4.0 standard geopotential z-coordinates
+"""
+
+
+import numpy as np
+from xarray import Dataset
+
+from .zgr import Zgr
+
+
+class Zco(Zgr):
+
+    # In future we may want to get rid of this
+    # and use a more pythonic way. We could do
+    # it when  we read the namelist.
+    pp_to_be_computed = 999999.0
+
+    """
+    Class to generate geopotential z-coordinates dataset objects.
+
+    Method
+    ------
+    *) Model levels' depths depT/W are defined from analytical function.
+    *) Model vertical scale factors e3 (i.e., grid cell thickness) can
+       be computed as
+
+             1) analytical derivative of depth function
+                (ln_e3_dep=False); for backward compatibility with v3.6.
+             2) discrete derivative (central-difference) of levels' depth
+                (ln_e3_dep=True). The only possibility from v4.0.
+
+    References:
+       *) NEMO v4.0 domzgr/zgr_z subroutine
+       *) Marti, Madec & Delecluse, 1992, JGR, 97, No8, 12,763-12,766.
+    """
+
+    # --------------------------------------------------------------------------
+    def __call__(
+        self,
+        ppdzmin,
+        pphmax,
+        ppkth: float = 0.0,
+        ppacr: int = 0,
+        ppsur: float = pp_to_be_computed,
+        ppa0: float = pp_to_be_computed,
+        ppa1: float = pp_to_be_computed,
```

Comment:
@oceandie 
My bad! If we change these to default None, we should also change the type annotation.
One option is:
```python
ppsur: Optional[float] = None
```
You would have to import Optional, so add `from typing import Optional` to the imports.

I think the docstring will need to be:
```
ppsur: float, optional
```

Reply:
The error was with that changes already implemented: 
def __call__(
        self,
        ppdzmin: float,
        pphmax: float, 
        ppkth: float = 0.0,
        ppacr: int = 0,
        ppsur: Optional[float] = None,
        ppa0: Optional[float] = None,
        ppa1: Optional[float] = None,
        ldbletanh: bool = False,
        ppa2: float = 0.0,
        ppkth2: float = 0.0,
        ppacr2: float = 0.0,
        ln_e3_dep: bool = True,
    ) -> Dataset:
        """
        Generate NEMO geopotential z-coordinates model levels.

        Parameters
        ----------
       .......
        ppsur: float, optional
            Coeff. controlling distibution of vertical levels
            (default = None, i.e. computed from ppdzmin, pphmax, ppkth and ppacr)
        ppa0: float, optional
            Coeff. controlling distibution of vertical levels
            (default = None, i.e. computed from ppdzmin, pphmax, ppkth and ppacr)
        ppa1: float, optional
            Coeff. controlling distibution of vertical levels
            (default = None, i.e. computed from ppdzmin, pphmax, ppkth and ppacr)

Reply:
Mmm not sure then, I'd have to take a look. Push the code anyways... When everything will be a bit more stable the whole pre-commit business will be much less annoying!